inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,516 @@
+        this->numbers_at_age[index_ya] =
+          this->numbers_at_age[index_ya] + 
+          this->numbers_at_age[index_ya2 + 1] *
+          (exp(-this->mortality_Z[index_ya2 + 1]));
+      }
+    }
+
+    /**
+     * @brief Calculates unfished numbers at age at year and age specific indices
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param index_ya2 dimension folded index for year-1 and age-1
+     * @param age age index
+     */
+    inline void CalculateUnfishedNumbersAA(int index_ya, int index_ya2, int age)
+    {
+      // using M from previous age/year - is this correct?
+      this->unfished_numbers_at_age[index_ya] =
+          this->unfished_numbers_at_age[index_ya2] *
+          (exp(-this->M[index_ya2]));
+
+           // Plus group calculation
+      if (age == (this->nages - 1)) {
+        this->unfished_numbers_at_age[index_ya] =
+          this->unfished_numbers_at_age[index_ya] + 
+          this->unfished_numbers_at_age[index_ya2 + 1] *
+          (exp(-this->M[index_ya2 + 1]));
+      }
+    }
+
+    /**
+     * @brief Calculates spawning biomass
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param year the year spawning biomass is being aggregated for
+     * @param age the age who's biomass is being added into total spawning biomass
+     */
+    void CalculateSpawningBiomass(int index_ya, int year, int age)
+    {
+      this->spawning_biomass[year] += this->proportion_female *
+                                      this->numbers_at_age[index_ya] *
+                                      // this->proportion_mature_at_age[age] *
+                                      this->proportion_mature_at_age[index_ya] *
+                                      this->weight_at_age[age];
+    }
+
+    /**
+     * @brief Adds to existing yearly unfished spawning biomass estimates the
+     *  biomass for a specified year and age
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param year the year of unfished spawning biomass to add
+     * @param age the age of unfished spawning biomass to add
+     */
+    void CalculateUnfishedSpawningBiomass(int index_ya, int year, int age)
+    {
+      this->unfished_spawning_biomass[year] += this->proportion_female *
+                                               this->unfished_numbers_at_age[index_ya] *
+                                               // this->proportion_mature_at_age[age] * Meg O made this change to match change in CalculateSpawningBiomass
+                                               this->proportion_mature_at_age[index_ya] *
+                                               this->weight_at_age[age];
+    }
+
+    /**
+     * @brief Calculates expected recruitment for a given year
+     *
+     * @param index_ya dimension folded index for year and age
+     * @param year the year recruitment is being calculated for
+     */
+    void CalculateRecruitment(int index_ya, int year)
```

Comment:
unit test on CalculateRecruitment currently not working and will be addressed during integration testing